gpytorch/distributions/multivariate_normal.py

```diff
@@ -114,6 +120,19 @@ def get_base_samples(self, sample_shape=torch.Size()):
             base_samples = _standard_normal(shape, dtype=self.loc.dtype, device=self.loc.device)
         return base_samples
 
+    @property
+    def base_sample_shape(self):
+        """
+        Returns the shape of a base sample (without batching) that is used to
+        generate a single sample.
+        """
+        base_sample_shape = self.event_shape
+        if isinstance(self.lazy_covariance_matrix, RootLazyTensor):
+            if self.lazy_covariance_matrix.root.shape[-1] > self.lazy_covariance_matrix.root.shape[-2]:
+                base_sample_shape = self.lazy_covariance_matrix.root.shape[-1:]
+
```

Comment:
maybe I'm missing something here, but what about the case where the root is a. tall tensor? Shouldn't `self.lazy_covariance_matrix.root.shape[-1:]` always be the `base_sample_shape`? If the reason is to allow easy dealing with varying posterior shapes for different test points (<= matrix size), this is probably something that we should deal with outside of the mvn to separate concerns better.

Reply:
It's already dealt with gracefully but the updated commit allows for it as well.